### DISCUSSION TITLE
SSD1306 OLED image suddenly garbled on screen

### DISCUSSION THREAD

**Discussion by probonopd** [probonopd/MiniDexed#579] [Q&A] [answered]
When browsing through performances using buttons attached to the RPi, my SSD1306 based 0.96" OLED went bad, kind of like the example "Image is Wrapped on the Screen" at https://iotexpert.com/debugging-ssd1306-display-problems/

Even restarting the device does not restore it.

Do we need an improved initialization sequence? Or did I encounter a hardware defect?

**Comment by diyelectromusic** [comment]
Nothing is really springing to mind.  But that talk of initialisation sequences on that web page implies to me that they are the different modes one could use with a ssd1306 display.  So they are an "all or nothing" setting - once set that is what the display will do.  For you to have gone from a working setting (so the init sequences were fine) to a non-working setup doesn't sound like the init sequences are at fault.

I think these displays remember their last contents until initialised again.  If so then it might be that no new commands are getting through - can you see the display resetting/reinitialising/updating (even if garbled) or is it completely frozen?

I guess it isn't impossible that something has happened to prevent the correct sequences being sent on startup.  But I suppose that could be at the display end, the RPi end or somewhere else on the I2C bus in between...

All I can suggest is substitution until you find something that is working again.

BTW - do you know if your display level shifts or not?  If not, I guess it is possible you might have caused an issue with the 3V3 I2C pins on the Pi, if they've been working with 5V levels...  If you're not sure, see the photos I added to the wiki a few weeks back, But to be sure, disconnect the display from the Pi, then power the display up and measure the voltage on the I2C pins to see if they're pulled up to 5V or 3V3.

It's worth checking before trying with a different Pi, just in case - don't want to risk frying a second one if it is that!

Kevin

**Reply by probonopd** [reply]
Thanks @diyelectromusic. The display still updates, it's just that the location of the content is shifted vs. where it should be. The whole thing had be running fine since >1 year in this setup, albeit the last 2 days saw a bit heavier usage. Will do the suggested checks when I find the time.

**Comment by probonopd** [comment, marked as answer]
Strangely, this seems to have fixed itself a few days later. Haven't done anything special besides leaving the device powered off for a while.
